### PR TITLE
test(stream): add backfill tests

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

We run `CREATE MATERIALIZED VIEW` concurrently with a workload of `INSERT`s + Barriers (triggered by `FLUSH`).
CI should fail in this branch, and then succeed after  fix from https://github.com/risingwavelabs/risingwave/pull/11082 is merged into this branch.

This is tested with and without deterministic simulation.
Deterministic simulation seems not to be able to reproduce in consistently. Probably some scheduling issue.
On the other hand running it externally via a bash script (w/o deterministic simulation) can reproduce it consistently.

This suggests that behaviour of backfill is different between the two test environments, so I will keep both tests.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by kwannoel** [comment]
Reproduced. https://buildkite.com/risingwavelabs/pull-request/builds/27803#018976d3-b49a-4f16-a952-08a1b1d48e6e

**Comment by kwannoel** [comment]
Now merging and testing it on latest.

File: ci/scripts/deterministic-it-test.sh [hunk 1]
```diff
@@ -5,6 +5,9 @@ set -euo pipefail
 
 source ci/scripts/common.sh
 
+echo "--- Running test for:"
+echo "$@"
+
```
Comment (kwannoel) [inline ci/scripts/deterministic-it-test.sh:10]:
```suggestion
```

**Review by chenzl25 — APPROVED** [review]
LGTM！

**Review by lmatz — APPROVED** [review]
LGTM

**Pull request merged**